src/components/ChannelList/ChannelListMessenger.tsx

```diff
@@ -152,19 +152,22 @@ export const ChannelListMessenger = <
   }, [loadingChannels]);
 
   const HeaderIndicator: React.FC = () => {
+    if (loadingChannels) return null;
+
     if (!isOnline) {
       return <HeaderNetworkDownIndicator />;
     } else if (error) {
       return <HeaderErrorIndicator onPress={refreshList} />;
     }
+
     return null;
   };
 
   const renderItem = (channel: Channel<At, Ch, Co, Ev, Me, Re, Us>) => (
     <ChannelPreview<At, Ch, Co, Ev, Me, Re, Us> {...props} channel={channel} />
   );
 
-  if (error && !refreshing && !channels?.length) {
+  if (error && !refreshing && !loadingChannels && !channels?.length) {
```

Comment:
Won't line 155 return early so no need to check loadingChannels?

Reply:
No ... line 155 is part of different function